mysql/datadog_checks/mysql/execution_plans.py

```diff
@@ -175,20 +175,70 @@ def _collect_execution_plans(self, db, tags, options):
                 num_truncated + len(events)
             )
 
-    def _run_explain(self, cursor, statement, schema):
+        return events
+
+    def _collect_execution_plans(self, db, tags, options, min_collection_interval):
+        if self._auto_enable_eshl is None:
+            self._auto_enable_eshl = is_affirmative(options.get('auto_enable_events_statements_history_long', False))
+        if not (is_dbm_enabled() and is_affirmative(options.get('collect_execution_plans', True))):
+            return False
+
+        stmt_row_limit = options.get('events_statements_row_limit', 5000)
+
+        events_statements_table = options.get('events_statements_table', 'events_statements_history_long')
```

Comment:
I don't think the config should have to specify which table to use. I think the integration should figure it out. Other suggestions: 

1. the query can do a UNION on both tables (although the checkpointing will have to be per-table).
2. on check setup, run through each table and check if it exists and is supported (i.e. is this an aurora reader)

Reply:
Yeah good point. I'll update to try them all automatically and look at the settings. 

Not sure about union though. I don't see a reason to query multiple sources. If the "long" history is enabled then no reason to query the short per thread history. Likewise, if we have short per thread, no reason to query the current table.